### PR TITLE
WIP: Integrate unipool

### PR DESCRIPTION
**Pull request by willjgriff**
WIP because I'm yet to successfully deploy to xDai.

Integrated deployment of a Unipool contract when using BYOT which can have the OG token sent to it as a reward for anyone holding the wrapped token. Note the Unipool is a token manager hook so when a user transfers the wrapped token the future reward for those tokens are passed to the receiver of the wrapped token.

Also added the voting aggregator so dao's can add other tokens which will be aggregated (typically added up) to determine vote weight used in the voting app.

Also added a second Agent for use with the voting app from which funds can't be distributed from conviction voting.

### CODE REVIEW

File: packages/hardhat/scripts/new-dao.ts [hunk 1]
```diff
@@ -229,23 +262,17 @@ export default async function main(log = console.log): Promise<any> {
...
-  if (!createNewToken) {
-    await approveOgtPayment(gardensTemplate, log)
-  }
-  const daoAddress = await createDaoTxOne(gardensTemplate, log)
-  if (createNewToken) {
-    await createTokenholders(gardensTemplate, log)
-  }
-  await createDaoTxTwo(gardensTemplate, log)
-  await createDaoTxThree(gardensTemplate, log)
+  // await approveHnyPayment(gardensTemplate, log)
```
Comment (0xGabi) [inline packages/hardhat/scripts/new-dao.ts:270]: We should remember to uncomment once we fix the price oracles.

**Review by 0xGabi — COMMENTED** [review]
Left a few comments. Thanks for adding the new changes, having a Unipool for BYOT is a great new feature.

File: packages/hardhat/contracts/GardensTemplate.sol [hunk 2]
```diff
@@ -170,25 +175,24 @@ contract GardensTemplate is BaseTemplate, AppIdsXDai {
...
             DisputableVoting disputableVoting,
             Agent commonPoolAgent,
-            IHookedTokenManager hookedTokenManager,
+            IHookedTokenManager hookedTokenManager
         ) = _getDeployedContractsTxOne();
 
-        if (hookedTokenManager.wrappableToken() == address(0)) { // if existingToken == address(0) (prevents stack too deep error)
-            _removePermissionFromTemplate(acl, hookedTokenManager, hookedTokenManager.MINT_ROLE());
+        if (!_creatingGardenWithExistingToken(hookedTokenManager)) {
+            _removePermissionFromTemplate(_getAcl(dao), hookedTokenManager, hookedTokenManager.MINT_ROLE());
```
[the remark this answers is not in the document]
Comment (willjgriff) [reply packages/hardhat/contracts/GardensTemplate.sol:203]: StackTooDeep error. Can't have any more references in the function.

**Review by 0xGabi — COMMENTED** [review]
Well done!

Wonder if we also want to consider using the Voting Aggregator for the Conviction Voting app. Doing a quick look at the contracts the `stakeToken` will be using the aggregated sources considering the configuration from the Voting Aggregator.

@sembrestels what you think?

**Pull request merged**